File: lab/javascript/textos/anagramas.js
function isAnagram(s1, s2) {
  if (s1.length !== s2.length) return false;
  const freq = Object.create(null);

  for (const ch of s1) {
    freq[ch] = (freq[ch] || 0) + 1;
  }
  for (const ch of s2) {
    if (!freq[ch]) return false;
    if (--freq[ch] === 0) delete freq[ch];
  }
  return Object.keys(freq).length === 0;
}

console.log(isAnagram("amor", "roma"));
console.log(isAnagram("casa", "caso"));

File: lab/python/textos/anagramas.py
def is_anagram(text1, text2):
    dict_contador = {}

    for text in text1:
        dict_contador[text] = dict_contador.get(text, 0) + 1

    for text in text2:
        if text not in dict_contador:
            return False
        dict_contador[text] -= 1

        if dict_contador[text] == 0:
            del dict_contador[text]

    return not dict_contador


print(is_anagram("amor", "roma"))
print(is_anagram("casa", "caso"))
